src/Agent.Worker/TaskRunner.cs

```diff
@@ -209,6 +209,23 @@ public async Task RunAsync()
                 // Expand the inputs.
                 Trace.Verbose("Expanding inputs.");
                 runtimeVariables.ExpandValues(target: inputs);
+
+                if (Task.Name.StartsWith("__system_posttargettask_")
+                    || Task.Name.StartsWith("__system_pretargettask_"))
```

Comment:
Could it make sense to export `__system_pretargettask_` and `__system_pretargettask_` as constants? Or add some field to `Task` that will allow us to determine if the task was injected from the decorator? 

Also what about other decorator's targets? I think we also should not provide access to secret variables for them as well. Or am I missing something?

Reply:
I have created an internal static class - _InjectedTasksNamesPrefixes_ to store these constants